libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
 	closedir (dd);
 	return ret;
 }
+
+
+/* Coredump  functions */
+
+static map_file_t mapping_file = {0,0};
+
+static char *prpsinfo_get_fname(FILE *f) {
+        char *p;
+        int c;
+        int pos;
+
+        p = malloc(16);
+        pos = 0;
+        while ((c = fgetc(f)) != EOF && c != '\0')
+                p[pos++] = c;
+
+        if (c == '\0')
+                p[pos] = c;
+
+        return p;
+}
+
+static char *get_basename(char *pfname, int len) {
+        char *p;
+
+        for (p = pfname + len; p != pfname ; p--)
+                if (*p == '/')
+                        return (p + 1);
+
+        return p;
+}
+
+static char *prpsinfo_get_psargs(FILE *f, char *pfname, int size_psargs) {
+        char paux[80];
+        char *p;
+        int c;
+        int pos;
+        int read;
+
+        p = malloc (size_psargs);
+
+        strcpy (p, pfname);
+
+        read = strlen (pfname);
+        pos = 0;
+	paux[pos++] = ' ';
+        while ((c = fgetc(f)) != EOF && read < (size_psargs - 1)) {
+                if (c == '\0')
+                        c = ' ';
+                paux[pos++] = c;
+                read++;
+        }
+
+        paux[pos] = '\0';
+        strcat (p, paux);
+
+        return p;
+}
+
+/*static void debug_print_prpsinfo(prpsinfo_t *p) {
+
+        printf("prpsinfo.pr_state: %d\n", p->pr_state);
+        printf("prpsinfo.pr_sname: %c\n", p->pr_sname);
+        printf("prpsinfo.pr_zomb: %d\n", p->pr_zomb);
+        printf("prpsinfo.pr_nice: %d\n", p->pr_nice);
+        printf("prpsinfo.pr_flags: %ld\n", p->pr_flag);
+        printf("prpsinfo.pr_uid: %ld\n", p->pr_uid);
+        printf("prpsinfo.pr_gid: %ld\n", p->pr_gid);
+        printf("prpsinfo.pr_pid: %ld\n", p->pr_pid);
+        printf("prpsinfo.pr_ppid: %ld\n", p->pr_ppid);
+        printf("prpsinfo.pr_pgrp: %ld\n", p->pr_pgrp);
+        printf("prpsinfo.pr_sid: %ld\n", p->pr_sid);
+        printf("prpsinfo.pr_fname: %s\n", p->pr_fname);
+	printf("prpsinfo.pr_psargs: %s\n", p->pr_psargs);
+}*/
+
+static prpsinfo_t *linux_get_prpsinfo(RDebug *dbg, proc_stat_content_t *proc_data) {
+        FILE *f;
+        char file[128];
+        long nice;
+        int c;
+        int pos;
+        pid_t mypid;
+	char *pfname;
+        char *ppsargs;
+        char *basename; /* pr_fname stores just the exec, withouth the path */
+	char prog_states[6] = "RSDTZW";				/* fs/binfmt_elf.c from kernel */
```

Comment:
Probably you don't need to use a fixed size, I think you can do:
```
char prog_states[] = "RSDTZW";
```
Also because you are then using string functions on this array of char, so it would be better to also store the terminating char, just to be safe